neo/Consensus/ConsensusService.cs

```diff
@@ -319,7 +322,7 @@ private void OnRecoveryMessageReceived(ConsensusPayload payload, RecoveryMessage
 
         private void OnPrepareRequestReceived(ConsensusPayload payload, PrepareRequest message)
         {
-            if (context.RequestSentOrReceived()) return;
+            if (context.RequestSentOrReceived() || context.ViewChanging()) return;
```

Comment:
You should allow receiving `PrepareRequest` and `PrepareResponse` even when `ViewChanging() == true`. Because `ViewChanging()` indicates that I want to change views, it may not be able to change successfully.

Reply:
See the comment below. accepting the prepare request could cause the node to commit after it has already changed view if the view is changing.

Reply:
This was burning our brain this afternoon, @erikzhang.  aehuaheuahuea
We got this error and stalling the CN. 
Someone used the changeview payload of a node to change to a higher view and then that node who sent the changeview payload committed.

At the beginning, as you may remember I was not very much in favor of not allowing views to move backward. I was supporting that committed nodes could recover nodes even in higher views.
But, at the end, I started to like this design. I think it was a nice standard to be followed.

In this, for being consistent with the change-view idea we may need to lock any state modification after changeview is requested. But we can still use the CN as a bridge, as in the log we presented as example in the beginning of the conversation of this thread.

However, we still needed to do a trick for the edge case, as you may see the flag `MoreThanFNodesCommitted`

Reply:
The flag used has been renamed and improvements have been made; resolving this now and awaiting further review.